### ISSUE TITLE
GetBEP20TransactionByAddress doesnt work when contractaddress is null

### ISSUE THREAD

**Issue by publee** [dannyhu/AtomicCore#2]
Hi, can you please fix api call to 

```
bscscanClient.GetBEP20TransactionByAddress(account, null, startblockblock, endblock)
https://api.bscscan.com/api?module=account&action=tokentx&address=0xABC&contractaddress=&page=1
```

it fails on "Error! Invalid contract address format" when contractaddress is empty or null

The proper implementation is to ommit "contractaddress=" entirely from url when is null or empty to make bscscan happy

Thanks.

**Comment by dannyhu** [comment]
I'll check your question later






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dannyhu/AtomicCore"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月23日(星期天) 凌晨0:59
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[dannyhu/AtomicCore] GetBEP20TransactionByAddress doesnt work when contractaddress is null (Issue #2)





 
Hi, can you please fix api call to
 bscscanClient.GetBEP20TransactionByAddress(account, null, startblockblock, endblock) https://api.bscscan.com/api?module=account&amp;action=tokentx&amp;address=0xABC&amp;contractaddress=&amp;page=1  
it fails on "Error! Invalid contract address format" when contractaddress is empty or null
 
The proper implementation is to ommit "contractaddress=" entirely from url when is null or empty to make bscscan happy
 
Thanks.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dannyhu** [comment]
update package to v1.0.3




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dannyhu/AtomicCore"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月23日(星期天) 凌晨0:59
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[dannyhu/AtomicCore] GetBEP20TransactionByAddress doesnt work when contractaddress is null (Issue #2)





 
Hi, can you please fix api call to
 bscscanClient.GetBEP20TransactionByAddress(account, null, startblockblock, endblock) https://api.bscscan.com/api?module=account&amp;action=tokentx&amp;address=0xABC&amp;contractaddress=&amp;page=1  
it fails on "Error! Invalid contract address format" when contractaddress is empty or null
 
The proper implementation is to ommit "contractaddress=" entirely from url when is null or empty to make bscscan happy
 
Thanks.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by publee** [comment]
It works!!! Thanks for quick fix.

**Comment by dannyhu** [comment]
It is recommended to use caching technology for some requests, and the method parameters are described, because BSC has a request frequency limit




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dannyhu/AtomicCore"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月24日(星期一) 凌晨2:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [dannyhu/AtomicCore] GetBEP20TransactionByAddress doesnt work when contractaddress is null (Issue #2)





 
Closed #2.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
